filebeat/prospector/prospectors/registry.go

```diff
@@ -0,0 +1,45 @@
+package prospectors
+
+import (
+	"fmt"
+
+	"github.com/elastic/beats/filebeat/channel"
+	"github.com/elastic/beats/filebeat/input/file"
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/libbeat/logp"
+)
+
+type ProspectorFactoryData struct {
+	States   []file.State
+	Done     chan struct{}
+	BeatDone chan struct{}
+}
+
+type ProspectorFactory func(config *common.Config, outletFactory channel.OutleterFactory, data ProspectorFactoryData) (Prospectorer, error)
```

Comment:
appropos naming, I will not complain if you rename `channel.OutleterFactory` to `channel.Factory` ;)